client/v3/config_test.go

```diff
@@ -118,13 +129,19 @@ func TestNewClientConfigWithSecureCfg(t *testing.T) {
 		t.Fatalf("Unexpected error: %v", err)
 	}
 
-	scfg := &clientv3.SecureConfig{
+	scfg := &SecureConfig{
 		Cert:   tls.CertFile,
 		Key:    tls.KeyFile,
 		Cacert: tls.TrustedCAFile,
 	}
 
-	cfg, err := newClientCfg([]string{"http://192.168.0.13:2379"}, 2*time.Second, 3*time.Second, 5*time.Second, scfg, nil)
+	cfg, err := NewClientConfig(&ConfigSpec{
+		Endpoints:        []string{"http://192.168.0.13:2379"},
+		DialTimeout:      2 * time.Second,
+		KeepAliveTime:    3 * time.Second,
+		KeepAliveTimeout: 5 * time.Second,
+		Secure:           scfg,
+	}, nil)
 	if cfg == nil || err != nil {
```

Comment:
About the test itself. You said that we cannot check the results using `expect.Equal`, however we can do better than just checking if it's not empty. I think we should check if `cfg.TLS` fields are not nil.

Reply:
updated.